File: visa-dev-samplecode-master/vdp-ruby/test/vta/manage_portfolios_test.rb
require 'test/unit'
require 'rest-client'
require 'json'
require 'yaml'
require File.expand_path('../../../lib/visa_api_client', __FILE__)

class ManagePortfoliosTest < Test::Unit::TestCase
  
  def setup
    @config = YAML.load_file('configuration.yml')
    @visa_api_client = VisaAPIClient.new
  end
  
  def test_portfolios
    base_uri = 'vta/'
    resource_path = "v3/communities/#{@config['vtaCommunityCode']}/portfolios"
    response_code = @visa_api_client.doMutualAuthRequest("#{base_uri}#{resource_path}", "Get Portfolio Details Test", "get", '', {'ServiceId' => @config['vtaServiceId']})
    assert_equal("200", response_code, "Get Portfolio Details test failed")
  end
end

File: visa-dev-samplecode-master/vdp-node/test/vta/manageportfoliostest.js
var request = require('request');
var fs = require('fs');
var config = require('../../config/configuration.json');
var VisaAPIClient = require('../../libs/visaapiclient.js');
var assert = require('chai').assert;
var randomstring = require('randomstring');

var req = request.defaults();
var userId = config.userId ;
var password = config.password;
var keyFile = config.key;
var certificateFile = config.cert;

describe('Manage Portfolios', function() {
	
	var visaAPIClient = new VisaAPIClient();
	
	it('Get Portfolio Details Test',function(done) {
		this.timeout(10000);
		var baseUri = 'vta/';
		var resourcePath = 'v3/communities/'+ config.vtaCommunityCode +"/portfolios";
		visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, '', 'GET', {'ServiceId' : config.vtaServiceId}, 
		function(err, responseCode) {
			if(!err) {
				assert.equal(responseCode, 200);
			} else {
				assert(false);
			}
			done();
		});
	});
});
